skbio/io/format/gff3.py

```diff
@@ -0,0 +1,508 @@
+    stream = StringIO(fh.read())
+    seq = read(stream, format='fasta', into=constructor, seq_num=seq_num)
+    seq.interval_metadata = _parse_record(lines, len(seq))
+    return seq
+
+
+def _yield_record(fh):
+    '''Yield (seq_id, lines) that belong to the same sequence.'''
+    lines = []
+    current = False
+    for line in _line_generator(fh, skip_blanks=True, strip=True):
+        if line.startswith('##sequence-region'):
+            _, seq_id, start, end = line.split()
+            length = int(end) - int(start) + 1
+            yield 'pragma', seq_id, length
+        if line.startswith('##FASTA'):
+            # stop once reaching to sequence section
+            break
+        if not line.startswith('#'):
+            try:
+                seq_id, _ = line.split('\t', 1)
+            except ValueError:
+                raise GFF3FormatError(
+                    'Wrong GFF3 format at line: %s' % line)
+            if current == seq_id:
+                lines.append(line)
+            else:
+                if current is not False:
+                    yield 'data', current, lines
+                lines = [line]
+                current = seq_id
+    yield 'data', current, lines
+
+
+def _parse_record(lines, length):
+    '''Parse the lines into a IntervalMetadata object.'''
+    interval_metadata = IntervalMetadata(length)
+    for line in lines:
+        columns = line.split('\t')
+        # there should be 9 columns
+        if len(columns) != 9:
+            raise GFF3FormatError(
+                'do not have 9 columns in this line: "%s"' % line)
+        # the 1st column is seq ID for every feature. don't store
+        # this repetitive information
+        metadata = {'source': columns[1],
+                    'type': columns[2],
+                    'score': columns[5],
+                    'strand': columns[6]}
+        phase = columns[7]
+        # phase value can only be int or '.'
+        try:
+            metadata['phase'] = int(phase)
+        except ValueError:
+            if phase != '.':
+                raise GFF3FormatError(
+                    'unknown value for phase column: {!r}'.format(phase))
+        metadata.update(_parse_attr(columns[8]))
+
+        start, end = columns[3:5]
+
+        bounds = [(int(start)-1, int(end))]
+
+        interval_metadata.add(bounds, metadata=metadata)
+
+    return interval_metadata
+
+
+def _parse_attr(s):
+    '''parse attribute column'''
+    voca_change = _vocabulary_change('gff3')
+    md = {}
+    # in case the line ending with ';', strip it.
+    s = s.rstrip(';')
+    for attr in s.split(';'):
+        k, v = attr.split('=')
+        if k in voca_change:
+            k = voca_change[k]
+        md[k] = v
+    return md
+
+
+def _serialize_interval_metadata(
+        interval_metadata, seq_id, fh, skip_subregion=True):
+    '''Serialize an IntervalMetadata to GFF3.
+
+    Parameters
+    ----------
+    interval_metadata : IntervalMetadata
```

Comment:
`seq_id` and `fh` aren't described. I realize this is a private method, but if the docstring goes into this level of detail it would be good to be complete